### ISSUE TITLE
Formatter mis-indents code

### ISSUE THREAD

**Issue by dmillard** [julia-vscode/julia-vscode#2059]
Sometimes VSCode formatting likes putting my `end` one indent too deep or too shallow. This happens for `struct`, `function`, and `for`.

If I can come up with a stand-alone example, I'll post it here.

**Comment by davidanthoff** [comment]
Yes, I've also noticed this, I think this regressed at some point. @ZacLN I think there are some auto format PRs that might exhibit that behavior in our repos.

**Comment by pfitzseb** [comment]
https://github.com/julia-vscode/julia-vscode/issues/1938 seems related.

**Comment by louie-github** [comment]
I think I can create a slightly different, but standalone example. This happens when using prefix notation to construct `Expr` objects:
```julia
:(
    function a(b, c, d)
        return b + c + d
    end
)
```

It gets indented as:
```julia
:(
    function a(b, c, d)
    return b + c + d
end
)
```

Sorry if this is too unrelated. I wasn't sure if this problem was different enough that a new issue would be needed.
